kroxylicious-operator/src/main/java/io/kroxylicious/kubernetes/operator/model/ingress/ProxyIngressModel.java

```diff
@@ -40,21 +40,31 @@ public Optional<VirtualClusterIngressModel> clusterIngressModel(VirtualKafkaClus
                 .findFirst();
     }
 
-    public record VirtualClusterIngressModel(VirtualKafkaCluster cluster, List<IngressModel> ingressModels) {
+    /**
+     *
+     * @param cluster cluster
+     * @param ingressModelResults the ingress model results, one per ingress in the VKC spec.ingresses in the same order
+     */
+    public record VirtualClusterIngressModel(VirtualKafkaCluster cluster, List<IngressModelResult> ingressModelResults) {
 
         public Stream<Service> services() {
-            return ingressModels.stream().flatMap(it -> it.ingressInstance().services()).map(ServiceBuilder::build);
+            return ingressModelResults.stream().flatMap(it -> it.ingressModel().services()).map(ServiceBuilder::build);
         }
 
         public Set<IngressConflictException> ingressExceptions() {
```

Comment:
Given 3 out of the 4 production usages just check if the resulting set is empty, I'm inclined to say we should be able to ask the the record if its valid or not and only use the this method in the one place that actually cares about the exception.

Reply:
Hmmmmmm am I contradicting my previous self with that. As validity for other result classes is often in the eye of the beholder. 

I feel like we really do want the `*Model` to be able to declare if its valid or not, but maybe not the results classes themselves.